cnc_ctrl_v1/Motor.cpp

```diff
@@ -87,28 +87,31 @@ void Motor::write(int speed){
         _lastSpeed = speed; //saves speed for use in additive write
         bool forward = (speed > 0);
         speed = abs(speed); //remove sign from input because direction is set by control pins on H-bridge   
+        
+        bool usePin1 = ((_pin1 != 4) && (_pin1 != 13) && (_pin1 != 11) && (_pin1 != 12)); // avoid PWM using timer0 or timer1
+        bool usePin2 = ((_pin2 != 4) && (_pin2 != 13) && (_pin2 != 11) && (_pin2 != 12)); // avoid PWM using timer0 or timer1
         if (forward) {
             if (speed > 0) {
-                if (_pin2 != 11) {          // avoid PWM using timer1
-                    digitalWrite(_pin1 , HIGH );
-                    analogWrite(_pin2 , 255 - speed); // invert drive signals - don't alter speed
-                } else {
-                    analogWrite(_pin1 , speed); // invert drive signals - don't alter speed
-                    digitalWrite(_pin2 , LOW );
-                }
-            } else {
-                digitalWrite(_pin1 , LOW );
+              if (usePin2) {
+                digitalWrite(_pin1 , HIGH );
+                analogWrite(_pin2 , 255 - speed); // invert drive signals - don't alter speed
+              } else {
+                analogWrite(_pin1 , speed);
                 digitalWrite(_pin2 , LOW );
+              }
+            } else {
+              digitalWrite(_pin1 , LOW );
+              digitalWrite(_pin2 , LOW );
             }
-        } else {
-            if (_pin1 != 11) {          // avoid PWM using timer1
```

Comment:
I am so glad we are getting rid of this special case if statement. It was ugly and your solution is so much cleaner 👍